lib/publish.ts

```diff
@@ -178,20 +203,22 @@ function doPublish(packageDir: string, gitRemoteUrl: string, params: Params): Pr
     }
 
     function commitChanges() {
-        const commitCommandText = `git commit --file="${commitTextPath}" --allow-empty-message --no-verify`;
-        return commitTextWritten.then(() => exec(commitCommandText, { cwd: gitRepoDir }));
+        const commitCommand = [`git`, `commit`, `--file="${commitTextPath}"`, `--allow-empty-message`, `--no-verify`];
+        return commitTextWritten.then(() => run(commitCommand, { cwd: gitRepoDir }));
     }
 
     function tagLastCommit() {
         return Promise.all([params.mainTagNameOp, tagTextWritten])
             .then(([tagName]) => {
-                return exec(`git tag -a --file="${tagTextPath}" "${tagName}"`, { cwd: gitRepoDir })
+                return parseOut(run([`git`, `tag`, `-a`, `--file="${tagTextPath}"`, `"${tagName}"`], { cwd: gitRepoDir }))
                 .then(() => {
                     let promises: Promise<string>[] = [];
                     (params.extraBranchNames || []).forEach((extraBranchName) => {
-                        promises.push(exec(
-                            `git branch -f "${extraBranchName}" "${tagName}"`,
+                        promises.push(parseOut(
+                          run(
+                            [`git`, `branch`, `-f`, `"${extraBranchName}"`, `"${tagName}"`],
```

Comment:
`"` make errors

Reply:
I have removed the quotes and reworked a little bit for synchronous execution.